opts/opts.go

```diff
@@ -182,27 +212,33 @@ func ValidateDevice(val string) (string, error) {
 //    [host-dir:]container-path[:rw|ro]
 // It will also validate the mount mode.
 func ValidatePath(val string) (string, error) {
-	return validatePath(val, true)
+	return validatePath(val, volumeMode)
 }
 
-func validatePath(val string, validateMountMode bool) (string, error) {
+func validatePath(val string, validateMode int) (string, error) {
```

Comment:
why don't you pass the validator function here instead of an int? It can be cleaner if we define this as:

```go
func validatePath(val string, validator func(string) (bool, error)) (string, error)
```

And then you call the validator directly:

```go
validator(mode)
```